drivers/shared/executor/executor_linux.go

```diff
@@ -145,11 +144,25 @@ func (l *LibcontainerExecutor) cleanOldProcessesInCGroup(nomadRelativePath strin
 			continue
 		}
 
-		err := syscall.Kill(pid, syscall.SIGKILL)
-		if err != nil {
-			l.logger.Error("unable to send signal to process", "pid", pid, "error", err)
+		if err := syscall.Kill(pid, syscall.SIGKILL); err != nil {
+			return fmt.Errorf("unable to send signal to process %d: %v", pid, err)
+		}
+	}
+
```

Comment:
If at this point `len(orphanedPIDS) == 0` I think we can just `return nil` without entering this loop.